sleap_nn/training/model_trainer.py

```
@@ -97,21 +98,37 @@
+            self.bin_files_path = self.dir_path
+
+        self.bin_files_path = f"{self.bin_files_path}/chunks_{datetime.strftime(datetime.now(), '%Y%m%d_%H-%M-%S-%f')}"
+        print(f"`.bin` files are saved in {self.bin_files_path}")
+
+        if not Path(self.bin_files_path).exists():
+            try:
+                Path(self.bin_files_path).mkdir(parents=True, exist_ok=True)
+            except OSError as e:
+                raise OSError(
+                    f"Cannot create a new folder in {self.bin_files_path}. Check the permissions to the given Checkpoint directory. \n {e}"
                 )
```

Ruff

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)